chemreg/substance/serializers.py

```diff
@@ -205,6 +208,18 @@ def validate(self, data):
             )
         return data
 
+    def get_display_name(self, obj):
+        display_name = obj.display_name or obj.preferred_name
+        if display_name == "":
+            display_name = None
+        return display_name
+
+    def save(self, **kwargs):
+        if "display_name" in self.initial_data:
+            kwargs.update(display_name=self.initial_data["display_name"])
+        super().save(**kwargs)
```

Comment:
It looks like this strategy bypasses the field validation for display name.  This is why blank display names are considered valid despite breaking the regex validator.